core/src/main/java/org/axonframework/messaging/annotation/EventAggregateIdentifier.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (c) 2010-2017. Axon Framework
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.messaging.annotation;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Annotation indication that a parameter on a Message Handler method should be
+ * injected with the aggregate identifier of a DomainEventMessage. The parameter type must be
+ * assignable from {@link String}.
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ElementType.PARAMETER, ElementType.ANNOTATION_TYPE})
+public @interface EventAggregateIdentifier {
```

Comment:
What about just `SourceId`? The current annotation name `EventAggregateIdentifier` kinda enforces it to be used for event messages only, whilst I can imagine this might be reusable for commands and queries too (albeit that the resolve would be more complicated).

Reply:
Sounds good, I changed it like that in the merge request.